sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -148,6 +147,86 @@ export {
   logger
 };
 
+/**
+ * Formats the received options into RequestOptions
+ */
+function toRequestOptionsBase(options?: CreateKeyOptions) {
+  let expandedOptions: RequestOptionsBase = {};
+
+  if (options) {
+    const requestOptions = operationOptionsToRequestOptionsBase(options);
+    const {
+      enabled,
+      notBefore,
+      expiresOn: expires,
+      hardwareProtected: hsm,
+      ...remainingOptions
+    } = requestOptions;
+    expandedOptions = {
+      ...remainingOptions,
+      keyAttributes: {
+        enabled,
+        notBefore,
+        expires,
+        hsm
+      }
+    };
+  }
+
+  return expandedOptions;
+}
+
+/**
+ * Prepares the request options and generates a tracer span.
+ * @param name Name of the tracer span.
+ * @param reqOptions Request options.
+ */
+function prepareTrace(name: string, reqOptions: RequestOptionsBase) {
+  const tracer = getTracer();
+  const span = tracer.startSpan(name, reqOptions && reqOptions.spanOptions);
+  span.setAttribute("az.namespace", "Microsoft.KeyVault");
+
+  let options = reqOptions;
+
+  // Setting the parent's span
+  if (span.isRecording()) {
+    const spanOptions = reqOptions.spanOptions || {};
+    options = {
+      ...reqOptions,
+      spanOptions: {
+        ...spanOptions,
+        parent: span.context(),
+        attributes: {
+          ...spanOptions.attributes,
+          "az.namespace": "Microsoft.KeyVault"
+        }
+      }
+    };
+  }
+
+  return { span, options };
+}
+
+/**
+ * Traces the given method, and returns the method's output without leaving active tracer spans.
+ * @param name Name of the tracer span.
+ * @param reqOptions Request options.
+ * @param method Method to wrap and trace.
+ */
+async function trace<T>(
+  name: string,
+  reqOptions: RequestOptionsBase,
+  method: (options: RequestOptionsBase) => Promise<T>
+) {
+  const tracing = prepareTrace(name, reqOptions);
+
+  try {
+    return await method(tracing.options);
+  } finally {
```

Comment:
what about the error case? you'd have to catch the error and mark the span as non-successful. Ideally you'd also be able to provide more details about the specific kind of error, which is hard to do generically.

Reply:
I see that for some reason we're not doing this in the client code today, which is concerning